### PR TITLE
Update circuitv2 to proto3

### PR DESCRIPTION
**Pull request by ckousik**
Updates circuit v2 to use proto3. Currently awaiting issue to be resolved and spec to be updated https://github.com/libp2p/specs/issues/509

### CODE REVIEW

**Comment by MarcoPolo** [comment]
Before merging we need to update the proto files now that https://github.com/libp2p/specs/pull/511 is in. Feel free to do so @ckousik, otherwise I'll pick this up tomorrow.

I think we are fine to merge this after we make the next release.

**Comment by p-shahi** [comment]
@ckousik are you picking this up or is Marco?

**Comment by ckousik** [comment]
@p-shahi I'm unable to add any reviewers.

**Comment by marten-seemann** [comment]
@ckousik Is this ready for review?

**Comment by ckousik** [comment]
@marten-seemann yes, just marked it ready for review

File: p2p/protocol/circuitv2/pb/voucher.proto
Comment (MarcoPolo) [inline p2p/protocol/circuitv2/pb/voucher.proto:6]: These need to be optional, no? That's what the specs say

**Review by MarcoPolo — CHANGES_REQUESTED** [review]

File: p2p/protocol/circuitv2/pb/voucher.proto
Comment (ckousik) [reply p2p/protocol/circuitv2/pb/voucher.proto]: Sorry, I missed this. Updated now to reflect https://github.com/libp2p/specs/pull/511/files?short_path=5d2c9cb#diff-5d2c9cb6bfb3274dc65770bf6735c8d6c343245aaf1d9fac474df154842b81d2

**Pull request closed without merging**
